### PR TITLE
[Dataset] Commenting (Issues ext Implementation)

### PR DESCRIPTION
**Pull request by steveoni**
This pr adds issue functionality to dataset page

Screenshot 2023-12-29 at 07 51 49
Screenshot 2023-12-29 at 07 52 03

### CODE REVIEW

**Comment by steveoni** [comment]
@luccasmmg when you are available re do you know why `reset()`  might not be working for the SimpleEditor?

**Comment by luccasmmg** [comment]
> @luccasmmg when you are available re do you know why `reset()` might not be working for the SimpleEditor?

The simpleeditor is a controlled component, meaning that the state is kept inside the component itself, https://react-hook-form.com/get-started#IntegratingControlledInputs i THINK it might work if you do this 

"For controlled components you will need to pass defaultValues to useForm in order to reset the Controller components' value."

So i think if you do `formObj.reset({description: ''})` it might work

File: deployment/frontend/src/server/api/routers/dataset.ts [hunk 1]
```diff
@@ -901,4 +935,170 @@ export const DatasetRouter = createTRPCRouter({
...
+                })
+            }
+            catch (error) {
+                console.log(error)
+                throw Error("Error in sending issue /comment notification")
+            }
+            return data
+        }),
+    
+    CloseOpenIssue: protectedProcedure
```
Comment (demenech) [inline deployment/frontend/src/server/api/routers/dataset.ts]: typo here in the name of the function (initial letter is uppercase)

**Review by demenech — CHANGES_REQUESTED** [review]
Overall it's working fine, just a few comments:

1) When rejecting and providing feedback, maybe the textarea should already be RTE
2) When adding a comment, if you add a link to a word and then click to open the context menu for the link, it's going to show up on a weird location
3) After rejecting I feel like the Approve/Reject buttons should disappear
4) The issues list should be a scrollable area with fixed height, to handle large quantities of issues

Just

**Comment by steveoni** [comment]
> Overall it's working fine, just a few comments:
> 
> 1. When rejecting and providing feedback, maybe the textarea should already be RTE
> 2. When adding a comment, if you add a link to a word and then click to open the context menu for the link, it's going to show up on a weird location
> 3. After rejecting I feel like the Approve/Reject buttons should disappear
> 4. The issues list should be a scrollable area with fixed height, to handle large quantities of issues
> 
> Just

@demenech I could not reproduce 2; about weird link opening. i added a link to the comment with text and I was able to click and go to the page.

>After rejecting I feel like the Approve/Reject buttons should disappear

no its not, the Reject button is not about just rejecting, is more about creating issues, and that's the only mean a user can use to create and add more issue. so till Approve is clicked all accessible user can always click reject to add more new issues

**Review by demenech — APPROVED** [review]
Lgtm

**Pull request merged**